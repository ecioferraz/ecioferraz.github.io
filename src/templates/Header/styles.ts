import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    align-items: center;
    background-color: rgba(0, 0, 0, 0.8);
    border-bottom: 1px solid ${theme.colors.primaryColor};
    box-shadow: ${theme.colors.secondaryColor} 0 3px 0 0,
      ${theme.colors.primaryColor} 0 3px 0 1px;
    box-sizing: border-box;
    display: flex;
    height: ${theme.spacings.xlarge};
    justify-content: space-between;
    overflow: auto;
    padding: 0 ${theme.spacings.xsmall};
    position: sticky;
    top: 0;

    svg {
      color: ${theme.colors.secondaryColor}
    }
  `}
`;

export const NavContainer = styled.nav`
  ${({ theme }) => css`
    align-items: center;
    display: flex;
    justify-content: space-around;

    > * {
      margin: 0 ${theme.spacings.xsmall};
      white-space: nowrap;
    }
  `}
`;

export const ContactContainer = styled.section`
  ${({ theme }) => css`
    align-items: center;
    display: flex;
    justify-content: center;

    > * {
      margin: 0 ${theme.spacings.xsmall};
    }
  `}
`;
