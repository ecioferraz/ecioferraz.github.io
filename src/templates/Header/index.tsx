import { Github, Linkedin } from '@styled-icons/boxicons-logos';
import CustomLink from '../../components/CustomLink';
import NavLink from '../../components/NavLink';
import * as Styled from './styles';

export default function Header() {
  return (
    <Styled.Container>
      <Styled.NavContainer>
        <ul>
          <NavLink src="#about">Quem sou eu</NavLink>
          <NavLink src="#skills">Habilidades</NavLink>
          <NavLink src="#projects">Projetos</NavLink>
        </ul>
      </Styled.NavContainer>
      <Styled.ContactContainer>
        <CustomLink src="https://github.com/ecioferraz">
          <Github />
        </CustomLink>
        <CustomLink src="https://www.linkedin.com/in/ecioferraz/">
          <Linkedin />
        </CustomLink>
      </Styled.ContactContainer>
    </Styled.Container>
  );
}
