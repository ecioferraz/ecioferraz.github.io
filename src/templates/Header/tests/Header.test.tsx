import renderTheme from '../../../styles/renderTheme';
import Header from '..';

describe('<Header />', () => {
  it('should render <Header />', () => {
    const { container } = renderTheme(<Header />);

    expect(container).toMatchSnapshot();
  });
});
