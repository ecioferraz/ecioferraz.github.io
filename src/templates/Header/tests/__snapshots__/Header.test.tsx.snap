// Vitest Snapshot v1

exports[`<Header /> > should render <Header /> 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.c4 svg {
  color: #000;
  width: 2rem;
}

.c4 svg:hover {
  scale: 1.3;
}

.c2 {
  background: none;
  cursor: pointer;
  color: #FFF;
  font-weight: 500;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2::before {
  border-bottom: 0.2rem solid rgba(255,255,255,0.5);
  bottom: 0;
  content: '';
  height: 100%;
  left: 0;
  opacity: 0;
  position: absolute;
  scale: 0.1,1;
  width: 100%;
  z-index: 1;
}

.c2:hover {
  -webkit-letter-spacing: 0.2rem;
  -moz-letter-spacing: 0.2rem;
  -ms-letter-spacing: 0.2rem;
  letter-spacing: 0.2rem;
}

.c2:hover::before {
  opacity: 1;
  scale: 1,1;
}

.c2::after {
  bottom: 0;
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c2:hover::after {
  opacity: 0;
  scale: 0.1,1;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(0,0,0,0.8);
  border-bottom: 1px solid #000;
  box-shadow: #FFF 0 3px 0 0, #000 0 3px 0 1px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 4.0rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow: auto;
  padding: 0 0.8rem;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
}

.c0 * {
  -webkit-transition: 0.8s all ease-in-out;
  transition: 0.8s all ease-in-out;
}

.c0 svg {
  color: #FFF;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c1 > * {
  margin: 0 0.8rem;
  white-space: nowrap;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 > * {
  margin: 0 0.8rem;
}

<div>
  <header
    class="c0"
  >
    <nav
      class="c1"
    >
      <a
        class="c2"
        href="#about"
        target="_self"
      >
        Quem sou eu
      </a>
      <a
        class="c2"
        href="#skills"
        target="_self"
      >
        Habilidades
      </a>
      <a
        class="c2"
        href="#projects"
        target="_self"
      >
        Projetos
      </a>
    </nav>
    <section
      class="c3"
    >
      <a
        class="c4"
        href="https://github.com/ecioferraz"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821 9.465.499.09.679-.217.679-.481 0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605 1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084 2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93 0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0 .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1 2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099 2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552 4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0 .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2 12.026 2z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
      <a
        class="c4"
        href="https://www.linkedin.com/in/ecioferraz/"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="4.983"
            cy="5.009"
            r="2.188"
          />
          <path
            d="M9.237 8.855v12.139h3.769v-6.003c0-1.584.298-3.118 2.262-3.118 1.937 0 1.961 1.811 1.961 3.218v5.904H21v-6.657c0-3.27-.704-5.783-4.526-5.783-1.835 0-3.065 1.007-3.568 1.96h-.051v-1.66H9.237zm-6.142 0H6.87v12.139H3.095z"
          />
        </svg>
      </a>
    </section>
  </header>
</div>
`;
